src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherV3Test.java

```diff
@@ -0,0 +1,238 @@
+package org.broadinstitute.dsde.consent.ontology.datause;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.broadinstitute.dsde.consent.ontology.model.DataUseV3;
+import org.broadinstitute.dsde.consent.ontology.model.DataUseBuilderV3;
+import org.broadinstitute.dsde.consent.ontology.model.TermParent;
+import org.broadinstitute.dsde.consent.ontology.model.TermResource;
+import org.broadinstitute.dsde.consent.ontology.service.AutocompleteService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+public class DataUseMatcherV3Test {
+
+  private static final String cancerNode = "http://purl.obolibrary.org/obo/DOID_162";
+  private static final String intestinalCancerNode = "http://purl.obolibrary.org/obo/DOID_10155";
+
+  @Mock
+  private AutocompleteService autocompleteService;
+
+  @BeforeEach
+  public void setUp() {
+    openMocks(this);
+    try {
+      when(autocompleteService.lookupById(any())).thenReturn(Collections.emptyList());
+    } catch (Exception e) {
+      //
+    }
+  }
+
+  @Test
```

Comment:
We need something to test abstentions. For example, this test generates a positive match, which is incorrect:
```
  @Test
  public void testAbstain() {
    DataUseV3 dataset = new DataUseBuilderV3().setGeneralUse(true).build();
    DataUseV3 purpose = new DataUseBuilderV3().setOther("Other").build();
    ImmutablePair<Boolean, List<String>> match = matchPurposeAndDataset(purpose, dataset);
    System.out.println(match);
  }
```
results in ...
```
(true,[])
```

In addition to this, I'm not sure our model will accommodate the abstain conditions we define in the algorithm doc. I think we could update the v3 schema for this. If we had a single field, `abstain` (or something like that), then when we're parsing the object that comes in from the resource class (not in this PR), then we could trigger that field if there were any unknown fields in the json.

Reply:
I'll start working on it!